packages/echo/src/client.test.ts

```diff
@@ -356,24 +358,52 @@ describe('Echo Client', () => {
         inputs: {},
       };
 
-      const executionResult = await echo.executeWorkflow(event);
-
-      expect(executionResult).toBeDefined();
-      expect(executionResult.outputs).toBeDefined();
-      if (!executionResult.outputs) throw new Error('executionResult.outputs is undefined');
+      const emailExecutionResult = await echo.executeWorkflow(emailEvent);
 
-      const body = (executionResult.outputs as any).body as string;
+      expect(emailExecutionResult).toBeDefined();
+      expect(emailExecutionResult.outputs).toBeDefined();
+      if (!emailExecutionResult.outputs) throw new Error('executionResult.outputs is undefined');
+      const body = (emailExecutionResult.outputs as any).body as string;
       expect(body).toBe('Test Body');
-
-      const subject = (executionResult.outputs as any).subject as string;
+      const subject = (emailExecutionResult.outputs as any).subject as string;
       expect(subject).toBe('Subject');
-
-      expect(executionResult.providers).toEqual({});
-
-      const metadata = executionResult.metadata;
+      expect(emailExecutionResult.providers).toEqual({});
+      const metadata = emailExecutionResult.metadata;
       expect(metadata.status).toBe('success');
       expect(metadata.error).toBe(false);
       expect(metadata.duration).toEqual(expect.any(Number));
+
+      const delayEvent: IEvent = {
+        action: 'execute',
+        data: {},
+        workflowId: 'test-workflow',
+        stepId: 'delay',
+        subscriber: {},
+        state: [
+          {
+            stepId: 'send-email',
+            outputs: {},
+            state: {
+              status: 'completed',
+              error: undefined,
+            },
+          },
+        ],
+        inputs: {},
+      };
+
+      const delayExecutionResult = await echo.executeWorkflow(delayEvent);
+
+      expect(delayExecutionResult).toBeDefined();
+      expect(delayExecutionResult.outputs).toBeDefined();
+      if (!delayExecutionResult.outputs) throw new Error('executionResult.outputs is undefined');
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion (non-blocking):** ‏ I would imagine only one of the 2 lines is required. I see this is the same as for the email step, if we choose to consolidate to one line I would do it in both places

Reply:
True only one is required the `if` statement is just a trick to help TypeScript understand that the variable has a value and won't be `null` later on. This lets us avoid adding question marks all over the code, making it cleaner to read (at least in my opinion).

Reply:
fair enough :)